File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if (!root) return []
    const arr = [root], result = []
    while (arr.length) {
        const size = arr.length, values = []
        for (let i = 0; i < size; i++) {
            const node = arr.shift()
            values.push(node.val)
            if (node.left) arr.push(node.left) 
            if (node.right) arr.push(node.right) 
        }
        result.push(values)
    }
    return result
};

File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
    result := [][]int{}
    if root == nil { return result }
    
    queue := []*TreeNode{root}
    
    for len(queue) > 0 {
        length := len(queue)
        level := []int{}
        
        for i := 0; i < length; i++ {
            node := queue[0]
            queue = queue[1:]
            level = append(level, node.Val)

            if node.Left != nil { 
                queue = append(queue, node.Left)
            }
            if node.Right != nil { 
                queue = append(queue, node.Right)
            }
        }
        result = append(result, level)
    }
    
    return result 
}